packages/core/src/app/ovm/deciders/and-decider.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  Decider,
+  Decision,
+  ImplicationProofElement,
+  Property,
+} from '../../../types/ovm/decider.interface'
+
+export interface AndDeciderInput {
+  left: Property
+  leftWitness: any
+  right: Property
+  rightWitness: any
+}
+
+/**
+ * Decider that decides true iff both of the provided properties evaluate to true.
+ */
+export class AndDecider implements Decider {
+  public async decide(
+    input: AndDeciderInput,
+    witness: undefined
+  ): Promise<Decision> {
+    const [leftDecision, rightDecision] = await Promise.all([
+      input.left.decider.decide(input.left.input, input.leftWitness),
+      input.right.decider.decide(input.right.input, input.rightWitness),
+    ])
+
+    if (!leftDecision.outcome) {
+      return leftDecision
+    }
+    if (!rightDecision.outcome) {
+      return rightDecision
+    }
+
+    const justification: ImplicationProofElement[] = []
+    if (!!leftDecision.justification.length) {
+      justification.concat(leftDecision.justification)
+    }
+    if (!!rightDecision.justification.length) {
+      justification.concat(rightDecision.justification)
+    }
```

Comment:
So we’ve commonly said that the `justification` will only be used when we decide `false`. This has been said because the case we were considering was when the on-chain decision differs from our local decision and we want to guard.

That said after looking at this it’s clear that when we want to send a coin & need to transfer a “proof” that this justification will come in very handy as it should be the entire proof that we are going to end up sending. Cool!

Reply:
Yeah, I tend to agree. To say that it's only necessary when we decide false is to say that we know how it's going to actually be used up the chain, and I don't think we do.